zulipterminal/ui_tools/messages.py

```diff
@@ -211,29 +214,33 @@ def top_search_bar(self) -> Any:
         else:
             self.model.controller.view.search_box.text_box.set_edit_text("")
         if curr_narrow == []:
-            text_to_fill = "All messages"
+            text_to_fill = f" {ALL_MESSAGES_MARKER} " + "All messages"
         elif len(curr_narrow) == 1 and curr_narrow[0][1] == "private":
-            text_to_fill = "All direct messages"
+            text_to_fill = f" {DIRECT_MESSAGE_MARKER} " + "All direct messages"
         elif len(curr_narrow) == 1 and curr_narrow[0][1] == "starred":
-            text_to_fill = "Starred messages"
+            text_to_fill = f" {STARRED_MESSAGES_MARKER} " + "Starred messages"
         elif len(curr_narrow) == 1 and curr_narrow[0][1] == "mentioned":
-            text_to_fill = "Mentions"
+            text_to_fill = f" {MENTIONED_MESSAGES_MARKER} " + "Mentions"
         elif self.message["type"] == "stream":
             assert self.stream_id is not None
+
             bar_color = self.model.stream_dict[self.stream_id]["color"]
             bar_color = f"s{bar_color}"
+            stream_access_type = self.model.stream_access_type(self.stream_id)
+            stream_icon = STREAM_ACCESS_TYPE[stream_access_type]["icon"]
             if len(curr_narrow) == 2 and curr_narrow[1][0] == "topic":
                 text_to_fill = (
                     "bar",  # type: ignore[assignment]
                     [
+                        (bar_color, f" {stream_icon} "),
                         (bar_color, self.stream_name),
                         (bar_color, ": topic narrow"),
                     ],
                 )
             else:
                 text_to_fill = (
                     "bar",  # type: ignore[assignment]
-                    [(bar_color, self.stream_name)],
+                    [(bar_color, f" {stream_icon} "), (bar_color, self.stream_name)],
```

Comment:
Similarly here, when the same style/attribute is applied, only one tuple is necessary, which means the strings can be combined.

You could also simplify the topic case above this in a similar way.